src/backend/distributed/metadata/node_metadata.c

```diff
@@ -234,6 +234,20 @@ citus_set_coordinator_host(PG_FUNCTION_ARGS)
 
 		UpdateNodeLocation(coordinatorNode->nodeId, nodeNameString, nodePort);
```

Comment:
I believe the reason we have this bug is that it's not obvious from naming or comments that UpdateNodeLocation does not sync metadata, while AddNodeMetadata does. Would be good to add comments and/or revise the name.

Reply:
I think it d be better if we let `UpdateNodeLocation` take an argument `localOnly`.